packages/hop-node/src/watchers/SyncWatcher.ts

```diff
@@ -104,6 +133,14 @@ class SyncWatcher extends BaseWatcher {
     )
   }
 
+  isAllSiblingWatchersSyncCompleted (syncIndex: number = this.syncIndex): boolean {
+    return Object.values(this.siblingWatchers).every(
+      (siblingWatcher: SyncWatcher) => {
+        return siblingWatcher.syncIndex === syncIndex
```

Comment:
I believe you're right. The problem is that isAllSiblingWatchersInitialSyncCompleted is only set to true once after initial sync but I need a way to know when every re-sync finishes to trigger the re-calculations and upload to s3. I'll think about how to do it better since it won't work based on what you said which I think you're correct